llvm/include/llvm/IR/Intrinsics.td

```diff
@@ -2367,6 +2367,10 @@ def int_hwasan_check_memaccess :
 def int_hwasan_check_memaccess_shortgranules :
   Intrinsic<[], [llvm_ptr_ty, llvm_ptr_ty, llvm_i32_ty],
             [ImmArg<ArgIndex<2>>]>;
+def int_hwasan_check_memaccess_fixedshadow_shortgranules :
```

Comment:
Can you document meaning of arguments.
I expected we are going to discard one of the arguments, like `int_asan_check_memaccess`.

Reply:
I'll add docs.

The way ASan avoids the need to pass the shadow offset via the intrinsic is by
defining llvm::getAddressSanitizerParams() in lib/Transforms/Instrumentation/AddressSanitizer.cpp. I'll try adapting that to this patch.

Reply:
Done

Reply:
To discard the arguments, I ended up adding getFixedShadowBase() to the HWAddressSanitizer pass